### PR TITLE
Adding references to supported postgresql libraries

### PR DESCRIPTION
**Pull request by javier**
Added docs about psycopg2, pg_connect, and ruby-pg libraries.

Also reworded more explicitly the link to Postgresql compatibility docs

### CODE REVIEW

File: docs/reference/api/postgres.md [hunk 1]
```diff
@@ -83,6 +83,13 @@ Support for `SELECT`, `INSERT`, `UPDATE`, `CREATE`, `DROP`, `TRUNCATE`.
 
 #### [pq](https://www.postgresql.org/docs/12/libpq.html) (C) `12`
 
+#### [Psycopg](https://www.psycopg.org/docs/) (Python) `2.9.3`
```
Comment (amyshwang) [inline docs/reference/api/postgres.md:86]: Based on my discussion with @amunra a while ago ([#999](https://github.com/questdb/questdb.io/issues/999), I think we now support psychopg3?

**Review by amyshwang — COMMENTED** [review]
One question!

File: docs/reference/api/postgres.md (hunk 1 above)
Comment (javier) [reply docs/reference/api/postgres.md:86]: I wasn't even aware psycopg3 was already released XD. Just tried it out and we seem to support it. Our Python exaples still show psycopg2, so maybe an opportunity to refresh the examples

Comment (javier) [reply docs/reference/api/postgres.md]: Just added the reference to psycopg3 on that page, but kept also the reference to psycopg2 as their official project page is still showing both

Comment (amyshwang) [reply docs/reference/api/postgres.md]: Yes #999 captures exactly that, the need to update the examples....

**Review by amyshwang — APPROVED** [review]

**Pull request merged**